x-template.js

```diff
@@ -870,9 +966,15 @@ class TemplateEngine {
   static #createRawResult(language, strings, values) {
     const analysis = TemplateEngine.#setIfMissing(TemplateEngine.#stringsToAnalysis, strings, () => ({}));
     if (!analysis.done) {
-      const fragment = TemplateEngine.#createFragment(language, strings);
-      const lookups = TemplateEngine.#findLookups(fragment);
-      analysis.language = language;
+      const lookups = {};
+      const onBoolean = TemplateEngine.#storeKeyLookup.bind(null, lookups, TemplateEngine.#BOOLEAN);
+      const onDefined = TemplateEngine.#storeKeyLookup.bind(null, lookups, TemplateEngine.#DEFINED);
+      const onAttribute = TemplateEngine.#storeKeyLookup.bind(null, lookups, TemplateEngine.#ATTRIBUTE);
+      const onProperty =  TemplateEngine.#storeKeyLookup.bind(null, lookups, TemplateEngine.#PROPERTY);
+      const onContent = TemplateEngine.#storeContentLookup.bind(null, lookups);
+      const onText = TemplateEngine.#storeTextLookup.bind(null, lookups);
+      const forgivingLanguage = language === TemplateEngine.#SVG ? Forgiving.svg : Forgiving.html;
+      const fragment = Forgiving.parse(strings, onBoolean, onDefined, onAttribute, onProperty, onContent, onText, forgivingLanguage);
```

Comment:
This is isn’t an interface we will ever make public. But — the idea is that it’s more like a visitor pattern. the parser just tells the integrator when something “interesting” happens and where it happened. This was one of the untangling changes we made for `Unforgiving` anyway, just back-porting it, really.